library/x509_crt.c

```diff
@@ -3823,10 +3858,19 @@ int mbedtls_x509_crt_verify_restartable( mbedtls_x509_crt *crt,
         return( ret );
     }
 
-    if( *flags != 0 )
-        return( MBEDTLS_ERR_X509_CERT_VERIFY_FAILED );
+    flags_fi = *flags;
+    if( flags_fi == 0 )
+    {
+        mbedtls_platform_enforce_volatile_reads();
+        if( flags_fi == 0 )
```

Comment:
Suggestion - 

Those checks will compile to be the same code. Probably a `CMP` or `TST` or maybe `TEQ` followed by a `BEQ`.

It would be nice to do the second check in a different way. That way, if the fault injection works on one instruction, the same fault can't be induced on the second. How about doing the second check as an xor? eg...
```
if(!(flags_fi ^ 0xffff))
{
   ...
```
That should compile to be `XOR` or `TEQ` followed by `BEQ`.

Alternatively, we could just try:
```
if(!flags_fi)
{
   ...
```